library/printoptions.mli

```diff
@@ -0,0 +1,42 @@
+(** Printing options record *)
+
+(* storage of the options is in the current instance of the record *)
+
+type t = {
+  printing_allow_match_default_clause : bool;
+  printing_coercions : bool;
+  printing_compact_contexts : bool;
+  printing_existential_instances : bool;
+  printing_factorizable_match_patterns : bool;
+  printing_implicit : bool;
+  printing_implicit_defensive : bool;
+  printing_matching : bool;
+  printing_notations : bool;
+  printing_primitive_projection_compatibility : bool;
+  printing_primitive_projection_parameters : bool;
+  printing_projections : bool;
+  printing_records : bool;
+  printing_synth : bool;
+  printing_universes : bool;
+  printing_wildcard : bool;
+}
+
+(** getter/setter for Printing options *)
+val get_current_options : unit -> t
+val set_current_options : t -> unit
```

Comment:
The boilerplate to get and set individual options may as well live in this module.

Reply:
Probably a good idea.